controllers/snapshot/snapshot_controller.go

```diff
@@ -96,10 +97,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 	logger = logger.WithApp(*application)
 
-	component, err := loader.GetComponentFromSnapshot(r.Client, ctx, snapshot)
+	var component *applicationapiv1alpha1.Component
+	err = retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
+		component, err = loader.GetComponentFromSnapshot(r.Client, ctx, snapshot)
+		if err != nil {
+			logger.Error(err, "Failed to get Component from the Snapshot")
+			return err
+		}
+		return nil
+	})
 	if err != nil {
-		logger.Error(err, "Failed to get Component from the Snapshot")
-		return ctrl.Result{}, err
+		if errors.IsNotFound(err) {
+			logger.Error(err, "Could not get component from Snapshot.  Component may have been removed.  Declining to proceed with reconciliation")
```

Comment:
We should IMHO just log at info level, it's expected that component can be removed while we keep snapshot CR present

Reply:
I kept it as an error to match the log levels of [these statements](https://github.com/redhat-appstudio/integration-service/pull/329/files/679cab5fd254920f4e2f0bbad72bbe1b3c6cf474#diff-bae1efbc7e547dded9da5f1294a79a891ee187e482dcb304bfe039b5302c7932R85) above it.  I think it's better for them to be consistent.

Reply:
those are errors, that state shouldn't ever happen, but here we are dealing with action that user intentionally removed component.